### DISCUSSION TITLE
DDL triggers do not work on changing the description of tables, fields, procedures, etc.

### DISCUSSION THREAD

**Discussion by Dtarabarov** [FirebirdSQL/firebird#8294] [Ideas]
Hi. 
This is really necessary.  Is this in the plans?)

**Comment by aafemt** [comment]
Is description considered metadata at all? Or it is just data?

**Comment by mrotteveel** [comment]
@Dtarabarov Can you edit this ticket to be more explicit, and provide a reproduction recipe that demonstrates the problem.

**Reply by Dtarabarov** [reply]
create table ddl_log (
       id bigint not null primary key,
       moment timestamp not null,
       user_name varchar(31) not null,
       event_type varchar(25) not null,
       object_type varchar(25) not null,
       ddl_event varchar(25) not null,
       object_name varchar(31) not null,
       sql_text blob sub_type text not null,
       ok char(1) not null
);

create sequence ddl_seq;

create or alter trigger trig_ddl_log_after
active after any ddl statement
as
begin
  insert into ddl_log (id, moment, user_name, event_type, object_type,
                       ddl_event, object_name, sql_text, ok)
  values (next value for ddl_seq, current_timestamp, current_user,
                   rdb$get_context('DDL_TRIGGER', 'EVENT_TYPE'),
                   rdb$get_context('DDL_TRIGGER', 'OBJECT_TYPE'),
                   rdb$get_context('DDL_TRIGGER', 'DDL_EVENT'),
                   rdb$get_context('DDL_TRIGGER', 'OBJECT_NAME'),
                   rdb$get_context('DDL_TRIGGER', 'SQL_TEXT'),
                   'Y');
end

Now, when metadata is changed, these changes will be recorded in the table ddl_log.
But when changing the metadata description, the trigger does not fire.
Different people write metadata descriptions. It is necessary to record who changes what..

**Reply by mrotteveel** [reply]
You mean using `COMMENT ON`?

**Reply by Dtarabarov** [reply]
Yes

**Comment by asfernandes** [comment]
COMMENT ON (as well GRANTS, for example), is documented to not fire the triggers.

**Reply by Dtarabarov** [reply]
Is it possible to make triggers be called? We really need to record changes in the metadata description.
